crates/spidermonkey-embedding-splicer/src/lib.rs

```diff
@@ -144,6 +148,11 @@ impl Guest for SpidermonkeyEmbeddingSplicerComponent {
             },
         )) = decode(&engine)
         {
+            // merge the imports from the engine with the imports from the guest content.
+            for (k, _) in &engine_resolve.worlds[engine_world].imports {
+                guest_imports.push(engine_resolve.name_world_key(k));
+            }
```

Comment:
This is actually causing the test case to fail. Specifically, in the test case even though the target world has all the imports, we are stubbing those imports, therefore they are being tree-shaken.

Instead, we should mark as live those imports which correspond to real imported imports in the engine _after stubbing_.